### ISSUE TITLE
[ERROR] Failed to execute goal net.ltgt.gwt.maven:gwt-maven-plugin:1.0-rc-9:generate-module (default-generate-module) on project project-client: Execution default-generate-module of goal net.ltgt.gwt.maven:gwt-maven-plugin:1.0-rc-9:generate-module failed: A required class was missing while executing net.ltgt.gwt.maven:gwt-maven-plugin:1.0-rc-9:generate-module: Lorg/sonatype/plexus/build/incremental/BuildContext;

### ISSUE THREAD

**Issue by valeriofinazzo** [tbroyer/gwt-maven-archetypes#53]
Hi,
i create my project using artifact modular-webapp. When i lunch mvn package i receive the error:
[ERROR] Failed to execute goal net.ltgt.gwt.maven:gwt-maven-plugin:1.0-rc-9:generate-module (default-generate-module) on project project-client: Execution default-generate-module of goal net.ltgt.gwt.maven:gwt-maven-plugin:1.0-rc-9:generate-module failed: A required class was missing while executing net.ltgt.gwt.maven:gwt-maven-plugin:1.0-rc-9:generate-module: Lorg/sonatype/plexus/build/incremental/BuildContext;
[ERROR] -----------------------------------------------------
[ERROR] realm =    extension>net.ltgt.gwt.maven:gwt-maven-plugin:1.0-rc-9
[ERROR] strategy = org.codehaus.plexus.classworlds.strategy.SelfFirstStrategy
[ERROR] urls[0] = file:/C:/Users/Valerio/.m2/repository/net/ltgt/gwt/maven/gwt-maven-plugin/1.0-rc-9/gwt-maven-plugin-1.0-rc-9.jar
[ERROR] urls[1] = file:/C:/Users/Valerio/.m2/repository/org/codehaus/plexus/plexus-utils/1.1/plexus-utils-1.1.jar
[ERROR] Number of foreign imports: 1
[ERROR] import: Entry[import  from realm ClassRealm[maven.api, parent: null]]
[ERROR]
[ERROR] -----------------------------------------------------
[ERROR] : org.sonatype.plexus.build.incremental.BuildContext
[ERROR] -> [Help 1]

i use the plugin version 1.0-rc-8 it works

**Comment by tbroyer** [comment]
Could you run with `mvn -X`?

You should see something like the following (here using Maven 3.5.2, OpenJDK 1.8.0_162); notice specifically the presence of `org.sonatype.plexus:plexus-build-api:jar:0.0.7:compile` which contains the class from your error message.
```
[DEBUG] net.ltgt.gwt.maven:gwt-maven-plugin:jar:1.0-rc-9:
[DEBUG]    org.apache.maven:maven-plugin-api:jar:3.0:compile
[DEBUG]       org.apache.maven:maven-model:jar:3.0:compile
[DEBUG]       org.apache.maven:maven-artifact:jar:3.0:compile
[DEBUG]       org.sonatype.sisu:sisu-inject-plexus:jar:1.4.2:compile
[DEBUG]          org.sonatype.sisu:sisu-inject-bean:jar:1.4.2:compile
[DEBUG]             org.sonatype.sisu:sisu-guice:jar:noaop:2.1.7:compile
[DEBUG]    org.sonatype.plexus:plexus-build-api:jar:0.0.7:compile
[DEBUG]    org.codehaus.plexus:plexus-archiver:jar:3.5:compile
[DEBUG]       org.codehaus.plexus:plexus-io:jar:3.0.0:compile
[DEBUG]       org.apache.commons:commons-compress:jar:1.14:compile
[DEBUG]       org.iq80.snappy:snappy:jar:0.4:compile
[DEBUG]       org.tukaani:xz:jar:1.6:runtime
[DEBUG]    org.codehaus.plexus:plexus-utils:jar:3.1.0:compile
[DEBUG]    org.codehaus.plexus:plexus-compiler-api:jar:2.8.2:compile
[DEBUG]    org.apache.maven:maven-core:jar:3.0:compile
[DEBUG]       org.apache.maven:maven-settings:jar:3.0:compile
[DEBUG]       org.apache.maven:maven-settings-builder:jar:3.0:compile
[DEBUG]       org.apache.maven:maven-repository-metadata:jar:3.0:compile
[DEBUG]       org.apache.maven:maven-model-builder:jar:3.0:compile
[DEBUG]       org.apache.maven:maven-aether-provider:jar:3.0:runtime
[DEBUG]       org.sonatype.aether:aether-impl:jar:1.7:compile
[DEBUG]          org.sonatype.aether:aether-spi:jar:1.7:compile
[DEBUG]       org.sonatype.aether:aether-api:jar:1.7:compile
[DEBUG]       org.sonatype.aether:aether-util:jar:1.7:compile
[DEBUG]       org.codehaus.plexus:plexus-interpolation:jar:1.14:compile
[DEBUG]       org.codehaus.plexus:plexus-classworlds:jar:2.2.3:compile
[DEBUG]       org.codehaus.plexus:plexus-component-annotations:jar:1.5.5:compile
[DEBUG]       org.sonatype.plexus:plexus-sec-dispatcher:jar:1.3:compile
[DEBUG]          org.sonatype.plexus:plexus-cipher:jar:1.4:compile
[DEBUG]    org.apache.maven:maven-archiver:jar:3.2.0:compile
[DEBUG]       org.apache.maven.shared:maven-shared-utils:jar:3.2.0:compile
[DEBUG]       commons-io:commons-io:jar:2.5:compile
[DEBUG]    org.apache.maven.surefire:maven-surefire-common:jar:2.20.1:compile
[DEBUG]       org.apache.maven.plugin-tools:maven-plugin-annotations:jar:3.3:compile
[DEBUG]       org.apache.maven.surefire:surefire-api:jar:2.20.1:compile
[DEBUG]          org.apache.maven.surefire:surefire-logger-api:jar:2.20.1:compile
[DEBUG]       org.apache.maven.surefire:surefire-booter:jar:2.20.1:compile
[DEBUG] Created new class realm extension>net.ltgt.gwt.maven:gwt-maven-plugin:1.0-rc-9
[DEBUG] Importing foreign packages into class realm extension>net.ltgt.gwt.maven:gwt-maven-plugin:1.0-rc-9
[DEBUG]   Imported:  < maven.api
[DEBUG] Populating class realm extension>net.ltgt.gwt.maven:gwt-maven-plugin:1.0-rc-9
[DEBUG]   Included: net.ltgt.gwt.maven:gwt-maven-plugin:jar:1.0-rc-9
[DEBUG]   Included: org.sonatype.sisu:sisu-inject-bean:jar:1.4.2
[DEBUG]   Included: org.sonatype.sisu:sisu-guice:jar:noaop:2.1.7
[DEBUG]   Included: org.sonatype.plexus:plexus-build-api:jar:0.0.7
[DEBUG]   Included: org.codehaus.plexus:plexus-archiver:jar:3.5
[DEBUG]   Included: org.codehaus.plexus:plexus-io:jar:3.0.0
[DEBUG]   Included: org.apache.commons:commons-compress:jar:1.14
[DEBUG]   Included: org.iq80.snappy:snappy:jar:0.4
[DEBUG]   Included: org.tukaani:xz:jar:1.6
[DEBUG]   Included: org.codehaus.plexus:plexus-utils:jar:3.1.0
[DEBUG]   Included: org.codehaus.plexus:plexus-compiler-api:jar:2.8.2
[DEBUG]   Included: org.sonatype.aether:aether-util:jar:1.7
[DEBUG]   Included: org.codehaus.plexus:plexus-interpolation:jar:1.14
[DEBUG]   Included: org.codehaus.plexus:plexus-component-annotations:jar:1.5.5
[DEBUG]   Included: org.sonatype.plexus:plexus-sec-dispatcher:jar:1.3
[DEBUG]   Included: org.sonatype.plexus:plexus-cipher:jar:1.4
[DEBUG]   Included: org.apache.maven:maven-archiver:jar:3.2.0
[DEBUG]   Included: org.apache.maven.shared:maven-shared-utils:jar:3.2.0
[DEBUG]   Included: commons-io:commons-io:jar:2.5
[DEBUG]   Included: org.apache.maven.surefire:maven-surefire-common:jar:2.20.1
[DEBUG]   Included: org.apache.maven.plugin-tools:maven-plugin-annotations:jar:3.3
[DEBUG]   Included: org.apache.maven.surefire:surefire-api:jar:2.20.1
[DEBUG]   Included: org.apache.maven.surefire:surefire-logger-api:jar:2.20.1
[DEBUG]   Included: org.apache.maven.surefire:surefire-booter:jar:2.20.1
```

You could try deleting `~/.m2/repository/net/ltgt/gwt/maven/gwt-maven-plugin/1.0-rc-9/`, it might be a corrupt download.

**Comment by valeriofinazzo** [comment]
ok. deleting plugin it works.
Reading log file previous release used org.sonatype.plexus:plexus-build-api:jar:0.0.4.
many thanks
